internal/kafka/internal/handlers/kafka_instance_types.go

```diff
@@ -0,0 +1,53 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/public"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/presenters"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/handlers"
+	"github.com/gorilla/mux"
+)
+
+type supportedKafkaInstanceTypesHandler struct {
+	service services.SupportedKafkaInstanceTypesService
+}
+
+func NewSupportedKafkaInstanceTypesHandler(service services.SupportedKafkaInstanceTypesService, cloudProviderService services.CloudProvidersService, kafkaService services.KafkaService, kafkaConfig *config.KafkaConfig) *supportedKafkaInstanceTypesHandler {
+	return &supportedKafkaInstanceTypesHandler{
+		service: service,
+	}
+}
+
+func (h supportedKafkaInstanceTypesHandler) ListSupportedKafkaInstanceTypes(w http.ResponseWriter, r *http.Request) {
+	cloudProvider := mux.Vars(r)["cloud_provider"]
+	cloudRegion := mux.Vars(r)["cloud_region"]
+
+	cfg := &handlers.HandlerConfig{
+		Validate: []handlers.Validate{
+			handlers.ValidateLength(&cloudProvider, "cloud_provider", &handlers.MinRequiredFieldLength, nil),
+			handlers.ValidateLength(&cloudRegion, "cloud_region", &handlers.MinRequiredFieldLength, nil),
+		},
+		Action: func() (i interface{}, serviceError *errors.ServiceError) {
+			supportedKafkaInstanceTypeList := public.SupportedKafkaInstanceTypesList{
+				InstanceTypes: []public.SupportedKafkaInstanceType{},
+			}
+
+			regionInstanceTypeList, err := h.service.GetSupportedKafkaInstanceTypesByRegion(cloudProvider, cloudRegion)
+			if err != nil {
+				return nil, err
```

Comment:
Not sure if we want to expose the InstanceTypeUnsupported error back to the client since this is something out of their control.

This error occurs when we have a mismatch in our configuration. We probably want to return a 500 error to the client and log/capture the InstanceTypeUnsupported error in sentry. wdyt?

Reply:
+1 on logging in sentry.
About the error code, maybe a 503 would be more appropriate?

Reply:
> About the error code, maybe a 503 would be more appropriate?

Why say `503 - Service Unavailable`, I was thinking more in line with what @JameelB suggested, `500 Internal Server Error` in this case.  Because the service is available, but an internal error due to config mismatch has occurred. Or we are saying, the error is to take the service down, my understanding is non due to configuration validation happening at app startup and technically this error will likely not occur during runtime.

Reply:
I'm unsure between 500 and 503, but I'm ok for 500 as well.

I was thinking 503 because the service, for that instance type is not available, but you can retry later.
While error 500 is about any "unexpected condition that prevented the server from fullfilling the request".

I'm interpreting 500 as something like 'network error', 'disk full', 'out of memory' or similar error that could happen at any time.

Reply:
> Not sure if we want to expose the InstanceTypeUnsupported error back to the client since this is something out of their control.

Would that affect also old clients that still want to use some unsupported instance types?
While we do not care about HTTP error codes, defining this error code might be beneficial for client to understand that they requesting some old type.

Reply:
I'll change the error reporting here. Thanks

Reply:
> > Not sure if we want to expose the InstanceTypeUnsupported error back to the client since this is something out of their control.
> 
> Would that affect also old clients that still want to use some unsupported instance types? While we do not care about HTTP error codes, defining this error code might be beneficial for client to understand that they requesting some old type.

Clients don't send a specific instance type in the request though, they're meant to get all of the instance types that KAS Fleet Manager supports. The only thing they can specify is the cloud provider id and region. If they requested an invalid cloud provider id/region then they'll get a service error code in relation to those.

This particular error is caused by misconfiguration on KAS Fleet Manager side. It will occur in this particular scenario: 
```
cloud provider config
---
supported_providers:
  - name: aws
    default: true
    regions:
      - name: us-east-1
        default: true
        supported_instance_type:
          standard: {}
          eval: {}
---

supported instance types config
---
supported_instance_types:
  - id: standard
    sizes:
       ...
---
```

`eval` is listed in `supported_instance_type` in us-east-1 but it's not available in the `supported_instance_types` configuration.

For this sort of error, maybe we could send back an `ErrorGeneral` service error (code: 9, http code: 500)? It's not something that clients can do anything about.